test/decl/typealias/generic.swift

```diff
@@ -328,7 +328,7 @@ typealias Y = ErrorA<ErrorB>
 
 typealias Id<T> = T
 
-extension Id {} // expected-error {{non-nominal type 'Id' cannot be extended}}
+extension Id {} // expected-error {{generic type 'Id' cannot be extended}}
```

Comment:
I wonder if there's some other message that could be used here instead. This reads to me with the connotation that generic types in general can't be extended, which obviously isn't true. What makes _this_ generic type such that it can't be extended?

Reply:
Oops, thats my mistake. I should’ve used “generic parameter” instead of “generic type”. Does that sound better though?

Reply:
I'm genuinely not sure what the diagnostic should say. I can write `typealias _Optional<T> = Optional<T>; extension _Optional { ... }` and I can write `typealias _Dictionary<T: Hashable> = Dictionary<T, T>; extension _Dictionary { ... }`. What's different about `Id`?

Reply:
Ah, I misunderstood. I think the core problem here is that we have an unbound generic type, so I think the error should say that we cannot extend this type without specifying what the generic parameter is.

Reply:
Okay, I think `extension Id` is equivalent to `extension Any` (which is disallowed) and `extension Id<Type>` (which is disallowed) is equivalent to `extension Id where T == Type` (which is also disallowed). I cannot suggest to the user do the former or latter and it seems to me that there is just no way to extend `Id` at all. It is, however, possible to extend `Type` directly, so perhaps I can say `generic type 'Id' cannot be extended; extension must be declared on a concrete type` or something similar (need to think more about wording).